### DISCUSSION TITLE
Custom Interface extension with an external chart vue component

### DISCUSSION THREAD

**Discussion by mark-james** [directus/directus#7454] [Archive]
Hi @rijkvanzanten, @nickrum,
I'm trying to get a basic chart displayed for a custom interface extension. Using apexchart and following the base vue example here https://apexcharts.com/docs/vue-charts/  it's not rendering. As the directus doco only provides the most basic of examples, I might be doing something incorrectly when bringing in an external component. 

My interface.vue file -

```vue
<template>
  <apexchart width="500" type="bar" 
  :options="{
        chart: {
          id: 'vuechart-example'
        },
        xaxis: {
          categories: [1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998]
        }
    }" 
  :series="[{
        name: 'series-1',
        data: [30, 40, 45, 50, 49, 60, 70, 91]
      }]">
  </apexchart>
</template>

<script>
import VueApexCharts from "vue3-apexcharts";

export default {
	 components: {
    	VueApexCharts
  	},
	props: {
		value: String,
	}
};
</script>
```

Index.js attached if that helps. Any direction would be great!

[index.js.zip](https://github.com/directus/directus/files/6999175/index.js.zip)

Thanks
Mark

**Comment by nickrum** [comment]
If you register a component locally by using

```js
components: {
  VueApexCharts
}
```

the name of the component inside the template has to match the name of the imported component. So, you have to use `<VueApexCharts>`-tags (`<vue-apex-charts>` should work as well) or rename the import.

**Reply by mark-james** [reply]
Thanks @nickrum!
That works. For other Vue newbies like me, you can also add an alias to the component, like this.

```
components: {
    	"apexchart": VueApexCharts
  	}
```

**Reply by Ayrcs** [reply]
Hello,
I’m interested in a chart interface extension.
Could you make your project public ?
Thanks
